src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -29,11 +29,11 @@ internal sealed class SynthesizedRecordPrintMembers : SynthesizedRecordOrdinaryM
 
         protected override DeclarationModifiers MakeDeclarationModifiers(DeclarationModifiers allowedModifiers, BindingDiagnosticBag diagnostics)
         {
-            var result = (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() && ContainingType.IsSealed) ?
+            var result = (ContainingType.IsRecordStruct || (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() && ContainingType.IsSealed)) ?
                 DeclarationModifiers.Private :
                 DeclarationModifiers.Protected;
 
-            if (virtualPrintInBase() is object)
+            if (ContainingType.IsRecord && virtualPrintInBase() is object)
```

Comment:
>virtualPrintInBase() [](start = 43, length = 20)

This logic doesn't feel right. If a record class doesn't inherit from object, the method should always have `Override` flag. If base doesn't contain the right method to override, an appropriate error is going to be reported about that.